be/src/olap/delta_writer.cpp

```diff
@@ -114,9 +114,8 @@ Status DeltaWriter::init() {
     // check tablet version number
     if (_tablet->version_count() > config::max_tablet_version_num) {
```

Comment:
Should we trigger a compaction before encountering an error. e.g.
if (_tablet->version_count() > config::max_tablet_version_num - 10) {
     // submit a compaction task?
}

Reply:
add a logic:
if (_tablet->version_count() > config::max_tablet_version_num - 100) trigger compaction. 

And I‘ve test this patch with flink checkpoint 1s and 20 concurrency stream load, compaction is under 40